### ISSUE TITLE
Issues with dumping video and audio

### ISSUE THREAD

**Issue by InfamousKnight** [hrydgard/ppsspp#10632]
What happens?
Some games work well and others act differently.

For instance: midnight club la remix works like a charm

And then other games that work odd: burnout legends, it won’t record video until an event begins I.e. race. Audio is recorded just fine

Star Wars battlefront ii: there seems to be some hiccups in the video. The audio still syncs with video, so there’s some hicccups there too

Final fantasy iv the complete collection acts strange too. When you load a save state, it doesn’t seem to renew the video part(from what I remember)

**Comment by FlatOutPS2** [comment]
I recently had a similar issue with Burnout Legends. The dumping didn't work at 30 FPS, but after I applied the 60 FPS patch it did work. After trying another 30 FPS game I noticed the same issue, while a native 60 FPS game was fine. It could be a coincidence, but it definitely suggests that lower framerates cause dumping issues.

**Comment by InfamousKnight** [comment]
60 frames per second patch fixed burnout legends? How is that done?

**Comment by LunaMoo** [comment]
So basically to sum this issue up, doesn't matter what framerate game runs, video records with the hardcoded 60000/1001 causing various weird stuff if the game doesn't actually feed those frames.

**Comment by unknownbrackets** [comment]
Hm.  Thing is, the PSP doesn't actually have the ability to display at less than 59.97 FPS.  It will simply display whatever is in VRAM at the display address every frame, no matter what speed the game renders at.

Of course, if the user selects "skip buffer effects" we won't emulate double buffering in games, and might have a broken frame.  But if they don't, `GPU_DBG_FRAMEBUF_DISPLAY` should in theory only select the same framebuffer that the PSP hardware would display.

It sounds like this means there's a bug in that, i.e. it's sometimes using the latched framebuf?

-[Unknown]

**Comment by LunaMoo** [comment]
GPU_DBG_FRAMEBUF_DISPLAY via it's description is the current display target, which doesn't equal to the displayed screen, so maybe it's not exactly a bug.

 I fixed the video recording by using output buffer and I think we still should at least have an option to record display with post process shaders, but as far as I tested now using render buffer also solves this issue, one downside - video will be in as huge resolution as the user selects and that becomes really slow really quickly.
 But at last it works better in software rendering and doesn't have the issue the output buffer had when recording that Darkstalkers game. I think I'll change my pull request to solve it via render buffer, but provide an option to record output buffer.
